fbw-a380x/src/wasm/systems/a380_systems/src/hydraulic/mod.rs

```diff
@@ -3227,6 +3261,9 @@ impl A380EngineDrivenPumpController {
         self.are_pumps_disconnected = self.are_pumps_disconnected
             || overhead_panel.engines_edp_disconnected(self.pump_id.into_engine_num());
```

Comment:
Maybe you should use `self.disconnection_mechanism.is_connected()` instead of `overhead_panel.engines_edp_disconnected(self.pump_id.into_engine_num())` here too (and move it after update)?

Reply:
Had a look and I think it would be different. In fact the mechanism.is_connected is the actual mechanical state of the mechanism, whereas self.are_pumps_disconnected is the logical state from cockpit/button point of view.
So if we have a failure in mechanism, you should then have self.are_pumps_disconnected = true while mechanism.is_connected staying false. I believe there's no actual feedback from the clutch system to the overhead, so my current understanding is that it can become out of sync